sympy/matrices/matrices.py

```diff
@@ -3433,6 +3433,99 @@ def _eval_matrix_exp_jblock(self):
         from .sparsetools import banded
         return self.__class__(banded(size, bands))
 
+
+
+    def analytic_func(self, f, x, parameter=None):
+        """
+        Computes f(A) where A is a Square Matrix
+        and f is an analytic function.
+
+        Examples
+        ========
+
+        >>> from sympy import Symbol, Matrix, exp, S, log
+
+        >>> x = Symbol('x')
+        >>> m = Matrix([[S(5)/4, S(3)/4], [S(3)/4, S(5)/4]])
+        >>> f = log(x)
+        >>> m.analytic_func(f, x)
+        Matrix([
+        [     0, log(2)],
+        [log(2),      0]])
+
+        Parameters
+        ==========
+
+        f : Analytic Function
+        x : parameter of f
+        parameter : If Symbol('t') then it returns f(A*t) where t is parameter
+
+        """
+        from sympy import eye, zeros, diff
+
+        if not self.is_square:
+            raise NonSquareMatrixError(
+                "Valid only for square matrices")
+
+        t = 1
+        if parameter:
+            t = sympify(parameter)
```

Comment:
I'd strictly check the parameter is `x` and `t` are `Symbol`

Reply:
`t` can be an expression, but x has to be a symbol.

Reply:
But what happens if `t` is an arbitrary expression? Does it give any intuitive result?

Reply:
For example, if x is also contained in parameter `m.analytic_func(log(x), x, x*t)`, it takes the derivative in different way

Reply:
I now would raise an another question whether the parameter `t` is needed or not
For example, you're necessarily doing the same thing like
`m.analytic_func(log(x*t), x)` than `m.analytic_func(log(x), x, t)`
I'd keep the API as minimal if some things are not necessary, and if the left thing can still be intuitive enough.

Reply:
I think x should not be in the parameter, there should be a check.

Reply:
Or it would be better if t is only a symbol, not an expr.

Reply:
But I mean that `m.analytic_func(log(x*t), x)` can be explained like `log(x*t)` become `log(X*t)`, 
so should that have been added as an additional parameter?

Reply:
in exp() there is no parameter option, so maybe this option can be removed make we make it similar to `exp()`, if a user want to make  parameter they could `(m*t).analytic_func(log(x), x)`, again here the symbol t should not be x.

Reply:
> But I mean that `m.analytic_func(log(x*t), x)` can be explained like `log(x*t)` become `log(X*t)`,
> so should that have been added as an additional parameter?

Yes i agree, i found that `simplify((m*t*sin(z)*cos(z)).analytic_func(log(x), x ))` and `simplify((m).analytic_func(log(x), x ,t*sin(z)*cos(z)))` there is a significant performance difference. It would be better to have a parameter

Reply:
Okay, so both looks like returning the equivalent result with first one more complicated.
- `(m*t).analytic_func(log(x), x)`
- `m.analytic_func(log(x*t), x)`

I think that the check can just be made easier that
1. `x` is a symbol
2. `x` belongs to `f`
3. `x` is not in `m`

Reply:
yes this is better, and parameter should be removed.

Reply:
The reason why the computation just becomes more slower for `(m*t).analytic_func(log(x), x)` is that you are multiplying a more complicated matrix containing `t` all over.

Reply:
`if x not in f.symbols`, can we just return f

Reply:
It may confuse the types, if users expect this to always return a matrix.
I see it currently returns identity matrix multiplied with `f`, but if you are uncertain about it, you can just raise an error.